mesonbuild/coredata.py

```diff
@@ -754,32 +750,33 @@ def process_compiler_options(self, lang: str, comp: Compiler, env: Environment,
 
         self.add_compiler_options(comp.get_options(), lang, comp.for_machine, env, subproject)
 
-        enabled_opts: T.List[OptionKey] = []
         for key in comp.base_options:
             if subproject:
                 skey = key.evolve(subproject=subproject)
             else:
                 skey = key
             if skey not in self.optstore:
-                self.optstore.add_system_option(skey, copy.deepcopy(compilers.base_options[key]))
+                self.optstore.add_system_option(skey.name, copy.deepcopy(compilers.base_options[key]))
```

Comment:
```suggestion
                self.optstore.add_system_option(skey, copy.deepcopy(compilers.base_options[key]))
```

From line above, we know that skey is not in self.optstore. Therefore, the set_value below cannot succeed if skey is a subproject option, because it will not find it in the optstore.